paddle/fluid/framework/channel_impl.h

```diff
@@ -118,15 +117,15 @@ bool ChannelImpl<T>::CanReceive() {
 }
 
 template <typename T>
-bool ChannelImpl<T>::Send(T *item) {
+void ChannelImpl<T>::Send(T *item) {
   send_ctr++;
   std::unique_lock<std::recursive_mutex> lock{mu_};
```

Comment:
Do you need to explicitly lock after constructor?  lock->lock() ?

Reply:
No we don't need to do that. The unique lock constructor automatically does that.